### DISCUSSION TITLE
StaticFiles returns a 5xx when filenames are too long

### DISCUSSION THREAD

**Discussion by etr2460** [encode/starlette#2221] [Potential Issue] [answered]
A filename that's too long causes an OSError, which StaticFiles doesn't catch and ends up as a 5xx. Ideally I think starlette should either:
- Catch OSErrors and return a 400 (Bad Request) instead
- Provide a way for a user to define a handle error function to handle this as expected

**Comment by etr2460** [comment]
I've implemented the first solution mentioned here in https://github.com/encode/starlette/pull/2222. Let me know if you think this is reasonable!

**Comment by Kludex** [comment, marked as answer]
I don't think it should be a 4xx. 🤔

You can inherit `StaticFiles`, and change the `get_response` implementation:

```py
from starlette.exceptions import HTTPException
from starlette.staticfiles import StaticFiles

class NewStaticFiles(StaticFiles):
    async def get_response(self, path: str, scope: Scope) -> Response:
        try:
            super().get_response(path, scope)
        except OSError as exc:
            raise HTTPException(status_code=400) from exc
```
